GiniVision/Classes/FilePickerManager.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  GalleryPickerManager.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 8/28/17.
+//  Copyright © 2017 Gini GmbH. All rights reserved.
+//
+
+import Foundation
+import MobileCoreServices
+
+internal final class FilePickerManager:NSObject {
+        
+    var didPickFile:((GiniVisionDocument) -> ()) = { _ in }
+    fileprivate var acceptedDocumentTypes = [kUTTypePDF as String] + GiniImageDocument.acceptedImageTypes
+    
+    // MARK: Picker presentation
+    
+    func showGalleryPicker(from:UIViewController) {
+        let imagePicker:UIImagePickerController = UIImagePickerController()
+        imagePicker.sourceType = .photoLibrary
+        imagePicker.delegate = self
+        from.present(imagePicker, animated: true, completion: nil)
+    }
+    
+    func showDocumentPicker(from:UIViewController) {
+        let documentPicker = UIDocumentPickerViewController(documentTypes: acceptedDocumentTypes, in: .open)
+        documentPicker.delegate = self
+        from.present(documentPicker, animated: true, completion: nil)
+    }
+    
+}
+
+// MARK: UIImagePickerControllerDelegate, UINavigationControllerDelegate
+
+extension FilePickerManager: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        if let pickedImage = info[UIImagePickerControllerOriginalImage] as? UIImage, let imageData = UIImageJPEGRepresentation(pickedImage, 1.0) {
+            let imageDocument = GiniImageDocument(data: imageData)
+            didPickFile(imageDocument)
+        }
+        
+        picker.dismiss(animated: true, completion: nil)
+    }
+    
+    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
+        picker.dismiss(animated: true, completion: nil)
+    }
+}
+
+// MARK: UIDocumentPickerDelegate
+
+extension FilePickerManager: UIDocumentPickerDelegate {
+    func documentPicker(_ controller: UIDocumentPickerViewController, didPickDocumentAt url: URL) {
+        do {
+            _ = url.startAccessingSecurityScopedResource()
+            let data = try Data(contentsOf: url)
+            url.stopAccessingSecurityScopedResource()
+            
+            if data.isPDF {
```

Comment:
Can you find a way to avoid these if statements? Like a factory method or something?